third_party/terraform/utils/provider_test.go.erb

```diff
@@ -96,6 +106,88 @@ func init() {
 	<% end -%>
 }
 
+var lock = &sync.Mutex{}
+
+// Returns a cached config if VCR testing is enabled. This enables us to use a single HTTP transport
+// for a given test, allowing for recording of HTTP interactions.
+func getCachedConfig(d *schema.ResourceData, configureFunc func(d *schema.ResourceData) (interface{}, error), testName string) (interface{}, error) {
+	lock.Lock()
+	defer lock.Unlock()
+	if _, ok := configs[testName]; !ok {
```

Comment:
nit: since there's only one line after the if statement, you could flip it and return early, saving some indentation in the rest of the statement. something like:
```
if v, ok := configs[testName]; ok {
  return v, nil
}
[everything else]
```